lib/sles4sap_publiccloud.pm

```diff
@@ -999,6 +1000,62 @@ sub saphanasr_showAttr_version {
     }
 }
 
+=head2 wait_for_cluster
+
+    Verifies that nodes are online, resources are started and DB is in sync
+
+=cut
+
+sub wait_for_cluster {
+    my ($self, $wait_time, $max_retries) = @_;
+
+    $wait_time //= 10;
+    $max_retries //= 7;
+
+    while ($max_retries > 0) {
+        my $hanasr_output = $self->run_cmd(cmd => 'SAPHanaSR-showAttr --format=script', quiet => 1);
+        my $crm_output = $self->run_cmd(cmd => 'crm_mon -r -R -n -N -1', quiet => 1);
+        my $resource_started = 1;    # assume resources are started at first
+        my $failed_actions = 0;
+
+        my $hanasr_ready = check_hana_topology(input => calculate_hana_topology(input => $hanasr_output));
+        my $crm_ok = check_crm_output(input => $crm_output);
+
+        # # Check nodes and sync state
```

Comment:
commented code to be deleted before to merge

Reply:
Added a `WIP` label to the PR to avoid merging by mistake with the commented code.